ablySpec/RealtimeClientConnection.swift

```diff
@@ -1302,6 +1302,53 @@ class RealtimeClientConnection: QuickSpec {
 
             }
 
+            // RTN15
+            context("connection failures once CONNECTED") {
+
+                // RTN15a
+                pending("should be able to catch up by receiving the queued ProtocolMessages if the connection is disconnected unexpectedly") {
+                    let options = AblyTests.commonAppSetup()
+
+                    let client1 = ARTRealtime(options: options)
+                    defer { client1.close() }
+                    let channel1 = client1.channels.get("test")
+
+                    let client2 = ARTRealtime(options: options)
+                    defer { client2.close() }
+                    let channel2 = client2.channels.get("test")
+
+                    let listener1 = channel1.subscribe { message in
+                        fail("Shouldn't receive the messsage")
+                    }
+
+                    expect(channel1.state).toEventually(equal(ARTRealtimeChannelState.Attached), timeout: testTimeout)
+
+                    client1.onDisconnected()
+                    expect(client1.connection.state).to(equal(ARTRealtimeConnectionState.Disconnected))
+
+                    channel2.publish(nil, data: "message") { errorInfo in
+                        expect(errorInfo).to(beNil())
+                    }
+
+                    waitUntil(timeout: options.disconnectedRetryTimeout) { done in
+                        client1.connection.on { stateChange in
+                            switch stateChange!.current {
+                            case .Connecting:
+                                channel1.unsubscribe(listener1)
```

Comment:
`listener1` should still work, shouldn't it? Maybe you can instead of failing inside unconditionally, add a condition that it has reached the reconnection. That way we test that previous subscriptions still work.